Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -5354,6 +5364,14 @@ void IFileContentManagerHost.ReportMaterializedArtifact(in FileOrDirectoryArtifa
             m_pipOutputMaterializationTracker.ReportMaterializedArtifact(artifact);
         }
 
+        private void MakeSharedOpaqueOutputIfNeeded(AbsolutePath path, bool force = false)
+        {
+            if (force || PipGraph.IsPathUnderOutputDirectory(path))

```

Comment:
Here I only check if the path is under an output directory; I could amend this check to additionally check if that directory is a **shared opaque** directory

Reply:
Scoping this down so we do this for shared opaques only makes sense to me. Sounds like the less intrusive change, considering that all this timestamp flagging is relatively hacky to being with.

---
In reply to: [272298426](https://github.com/Microsoft/BuildXL/pull/83#discussion_r272298426) [](ancestors = 272298426)

Reply:
agreed